app/js/trouble.js

```diff
@@ -1,123 +1,104 @@
-'use strict';
+(function(gbl, $) {
+  'use strict';
+  gbl.dropdown = function(options) {
+    var $dropdown = options.dropdown;
+    var $toggle = options.toggleButton;
+    var ns = options.namespace;
+    var controlsMegaMenu = (options.controlsMegaMenu && options.controlsMegaMenu ===
+      false) ? false : true;
+    var clickAnywhereToClose = (options.clickAnywhereToClose && options.clickAnywhereToClose ===
+      false) ? false : true;
+    var transitionEnd = gbl.utilities.whichTransitionEvent();
+    $dropdown.addClass('gbl_dropdown').data('status', 'closed');
+    $toggle.addClass('gbl_dropdown_trigger');
+    $dropdown.attr('aria-expanded', 'false');
+    $toggle.attr('aria-controls', $dropdown.attr('id'));
+    $dropdown.wrapInner('<div class="measureHeight"></div>');
 
-(function (gbl, $) {
-	
-	gbl.dropdown = function (options) {
-		
-		var $dropdown = options.dropdown,
-			$toggle = options.toggleButton,
-			token = +new Date(),
-			ns = options.namespace
-			elList = $('.dropdowns'),
-			controlsMegaMenu = (options.controlsMegaMenu && options.controlsMegaMenu == false)? false: true;
-			clickAnywhereToClose = (options.clickAnywhereToClose && options.clickAnywhereToClose == false) ? false : true,
-			transitionEnd = gbl.utilities.whichTransitionEvent();
+    function closeMegaMenu() {
+      if (gbl.megaMenu && gbl.megaMenu.small && typeof gbl.megaMenu.small
+        .close == 'function') {
+        gbl.megaMenu.small.close();
+    }
+  }
 
+  function setCloseHandler() {
+    $(document).on('click.' + ns, function(e) {
+      var $clicked = $(e.target);
+      if (!$clicked.is($dropdown) && ($clicked.parents().filter(
+        $dropdown).length === 0)) {
+        close();
+    }
+  });
+  }
 
-		$dropdown.addClass('gbl_dropdown').data('status', 'closed');
-		$toggle.addClass('gbl_dropdown_trigger')
-		$dropdown.attr('aria-expanded', 'false')
-		$toggle.attr('aria-controls', $dropdown.attr('id'));
-		$dropdown.wrapInner('<div class="measureHeight"></div>');
+  function removeCloseHandler() {
+    $(document).off('click.' + ns);
+  }
 
-		function closeMegaMenu() {
-			if (gbl.megaMenu && gbl.megaMenu.small && typeof gbl.megaMenu.small.close == 'function') {
-				gbl.megaMenu.small.close();
-			}
-		}
+  function setDropdownHeight() {
+    $dropdown.height($dropdown.find('.measureHeight').height());
+  }
 
-		function setCloseHandler() {
-			$(document).on('click.' + ns, function (e) {
-				var $clicked = $(e.target)
-				if (!$clicked.is($dropdown) && ($clicked.parents().filter($dropdown).length == 0)) {
-					close();
-				}
-			});
-		}
+  function close() {
+    var i;
+    setDropdownHeight();
+    $dropdown.data('status', 'closed');
+    $dropdown.removeClass('gbl_dropdown_active');
+    Array.prototype.foreach = function(cb) {
+      for (i = 0; i < this.length; ++i) {
+        cb(this[i]);
+      }
+    };
+    while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this does remove "making" a function in a loop, the desire here is to either move the anonymous function outside the loop and assign it to a variable, passing in the variable name as the first parameter to setTimeout, or simply declaring the function outside the loop and passing in the function name as that first parameter. By invoking the function here, it will call the function as soon as the code executes, rather than in response to the timeout event later.